db/commitlog/commitlog.cc

```diff
@@ -930,9 +932,8 @@ class db::commitlog::segment : public enable_shared_from_this<segment>, public c
             auto finally = defer([&] () noexcept {
                 _segment_manager->notify_memory_written(size);
                 _segment_manager->totals.buffer_list_bytes -= buf.size_bytes();
-                if (_size_on_disk < _file_pos) {
-                    _segment_manager->totals.total_size_on_disk += (_file_pos - _size_on_disk);
-                    _size_on_disk = _file_pos;
```

Comment:
I appreciate the better consistency, but this is adding an allocation+task to the hot path. It's not quite per write, but per (small number of writes), for the more durable tables.

Reply:
(per write == per user INSERT/UPDATE, not per commitlog write)

Reply:
It is not really the "hot" path... And write is already nestled with continuations (less in refactor coming branch, but...). Seems to me that is is not that hefty an addition.

Reply:
Let's see perf_simple_query with commitlog in batch mode (on ramdisk).

Reply:
For some reason I cannot run `perf_simple_query` on ramdisk.... Keep getting `EINVAL`...
I also had to modify the benchmark to add a commitlog batch option...

Anyway, with plain old tmpfs, we get:
master:
```
	{
		"allocs_per_op" : 68.29561511944604,
		"instructions_per_op" : 0.0,
		"mad tps" : 2118.3316244413145,
		"max tps" : 425416.49492983491,
		"median tps" : 397582.47153378825,
		"min tps" : 395464.13990934694,
		"tasks_per_op" : 18.731985105677328
	},
	{
		"allocs_per_op" : 68.259809725211042,
		"instructions_per_op" : 0.0,
		"mad tps" : 13631.479232312704,
		"max tps" : 420582.24925546983,
		"median tps" : 402449.35569910001,
		"min tps" : 385442.43090700504,
		"tasks_per_op" : 18.712286946097084
	},
	{
		"allocs_per_op" : 68.289302249156691,
		"instructions_per_op" : 0.0,
		"mad tps" : 3096.7571222148254,
		"max tps" : 453216.03738224204,
		"median tps" : 413443.41797986667,
		"min tps" : 397973.08724443399,
		"tasks_per_op" : 18.730835773866481
	},

```

PR:
```
	{
		"allocs_per_op" : 68.385908071775702,
		"instructions_per_op" : 0.0,
		"mad tps" : 9420.6814274010831,
		"max tps" : 466017.08227707446,
		"median tps" : 417543.57998778007,
		"min tps" : 396314.56719759374,
		"tasks_per_op" : 18.80612711258085
	},
	{
		"allocs_per_op" : 68.090368586471001,
		"instructions_per_op" : 0.0,
		"mad tps" : 18115.507142089424,
		"max tps" : 465545.4676978676,
		"median tps" : 425675.20211426448,
		"min tps" : 402974.35145419918,
		"tasks_per_op" : 18.621874361845208
	},
	{
		"allocs_per_op" : 68.119316276455564,
		"instructions_per_op" : 0.0,
		"mad tps" : 24642.845125292952,
		"max tps" : 456051.16822472482,
		"median tps" : 422974.69748250721,
		"min tps" : 388181.71933821432,
		"tasks_per_op" : 18.638215970867595
	},
```

No IPO, since all was run inside docker unfortunately. Not very super stable, but allocs and tasks per op seems to be pretty en par between the two.

Reply:
I did mean tmpfs (do ramdisks still exist? what did you use?)

Looks like your docker ate the performance monitor support so we don't see instructions per op. But tasks and allocations look reasonable.

Reply:
ramfs vs. tmpfs. And yes, docker kills PM:s, tried to pass it through before but never gotten it to work 100%. 
In any case, even in batch, there are so many continuations involved that one more should make very little difference. And this PR is also meant to lay some groundwork for bigger refactor that tries to slim said write path a bit...